### PR TITLE
xctesthtmlreport 2.2.2

### PR DESCRIPTION
**Pull request by tylervick**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/xctesthtmlreport.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Xctesthtmlreport < Formula
+  desc "Xcode-like HTML report for Unit and UI Tests"
+  homepage "https://github.com/XCTestHTMLReport/XCTestHTMLReport"
+  url "https://github.com/XCTestHTMLReport/XCTestHTMLReport/archive/refs/tags/2.2.2.tar.gz"
+  sha256 "1d7159e4e66feae6194bec2303690e91db17e2593b597ac0db357942020bf4a9"
+  license "MIT"
+  head "https://github.com/XCTestHTMLReport/XCTestHTMLReport.git", branch: "main"
+
+  depends_on xcode: "13.0"
```
Comment (carlocab) [inline Formula/xctesthtmlreport.rb:9]:
```suggestion
  depends_on xcode: "13.0"
  uses_from_macos "swift"
```
Is Xcode/Swift needed at runtime?

File: Formula/xctesthtmlreport.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    system "swift", "build", "--disable-sandbox", "-c", "release"
+    bin.install ".build/release/xchtmlreport"
+  end
+
+  test do
+    assert_match "XCTestHTMLReport #{version}", shell_output("#{bin}/xchtmlreport -h | head -n 1")
```
Comment (carlocab) [inline Formula/xctesthtmlreport.rb:17]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

-----

You also don't need to pipe to `head` when doing `assert_match`.

File: Formula/xctesthtmlreport.rb (hunk 1 above)
Comment (tylervick) [reply Formula/xctesthtmlreport.rb:9]: Yes, the tool calls `xcrun`, which is installed with Xcode Command Line Tools. Swift isn't required unless building from source - should it still be listed here?

File: Formula/xctesthtmlreport.rb (hunk 2 above)
Comment (tylervick) [reply Formula/xctesthtmlreport.rb]: Thank you! I added a more comprehensive functional test (and removed the unnecessary pipe).

File: Formula/xctesthtmlreport.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/xctesthtmlreport.rb:10]: Does it not need a swift runtime?

File: Formula/xctesthtmlreport.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  homepage "https://github.com/XCTestHTMLReport/XCTestHTMLReport"
+  url "https://github.com/XCTestHTMLReport/XCTestHTMLReport/archive/refs/tags/2.2.2.tar.gz"
+  sha256 "1d7159e4e66feae6194bec2303690e91db17e2593b597ac0db357942020bf4a9"
+  license "MIT"
+  head "https://github.com/XCTestHTMLReport/XCTestHTMLReport.git", branch: "main"
+
+  depends_on :macos
+  depends_on xcode: "13.0"
+
+  resource "testdata" do
```
Comment (SMillerDev) [inline Formula/xctesthtmlreport.rb:12]:
```suggestion
  resource "homebrew-testdata" do
```

File: Formula/xctesthtmlreport.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 "ce574435d6fc4de6e581fa190a8e77a3999f93c4714582226297e11c07d8fb66"
+  end
+
+  def install
+    system "swift", "build", "--disable-sandbox", "-c", "release"
+    bin.install ".build/release/xchtmlreport"
+  end
+
+  test do
+    resource("testdata").stage("SanityResult.xcresult")
```
Comment (tylervick) [inline Formula/xctesthtmlreport.rb:23]:
```suggestion
    resource("homebrew-testdata").stage("SanityResult.xcresult")
```

**Pull request closed without merging**